src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconService.kt

```diff
@@ -69,19 +69,27 @@ class ActivitiesReconService(
     )
   }
 
-  private fun publishTelemetry(
+  private suspend fun publishTelemetry(
     prisonId: String,
     compareResults: CompareBookingsResult,
   ) {
     val telemetryName =
       "activity-allocation-reconciliation-report-${if (compareResults.noDifferences()) "success" else "failed"}"
-    val telemetryMap = mutableMapOf("prison" to prisonId)
-
-    if (compareResults.nomisOnly.isNotEmpty()) telemetryMap["NOMIS_only"] = compareResults.nomisOnly.toString()
-    if (compareResults.dpsOnly.isNotEmpty()) telemetryMap["DPS_only"] = compareResults.dpsOnly.toString()
-    if (compareResults.differentCount.isNotEmpty()) telemetryMap["different_count"] = compareResults.differentCount.toString()
+    if (compareResults.noDifferences()) {
+      telemetryClient.trackEvent(telemetryName, mapOf("prison" to prisonId))
+    }
 
-    telemetryClient.trackEvent(telemetryName, telemetryMap.toMap())
+    val bookingDetails = nomisApiService.getPrisonerDetails(compareResults.nomisOnly + compareResults.dpsOnly + compareResults.differentCount)
+    bookingDetails
+      .sortedBy { it.bookingId }
+      .forEach {
+        val telemetryMap = mutableMapOf("prison" to prisonId)
+        telemetryMap["type"] = compareResults.differenceType(it.bookingId)
+        telemetryMap["bookingId"] = it.bookingId.toString()
+        telemetryMap["offenderNo"] = it.offenderNo
+        telemetryMap["location"] = it.location
+        telemetryClient.trackEvent(telemetryName, telemetryMap.toMap())
+      }
```

Comment:
We now publish a separate telemetry event for each failure including some details to assist with identifying the issue (99% of the time these are due to an incorrect location). Including offenderNo because this will then appear on a general App Insights query by offender (including all sync messages).